src/bin/upgrade/main.rs

```diff
@@ -29,14 +26,64 @@ use termcolor::{BufferWriter, Color, ColorChoice, ColorSpec, WriteColor};
 use url::Url;
 
 mod errors {
-    error_chain! {
-        links {
-            CargoEditLib(::cargo_edit::Error, ::cargo_edit::ErrorKind);
+    use thiserror::Error as ThisError;
+
+    #[derive(Debug, ThisError)]
+    pub enum Error {
+        /// An error originating from the cargo-edit library
+        #[error(transparent)]
+        CargoEditLib(#[from] ::cargo_edit::Error),
```

Comment:
I think we're using edition 2018, so `::` prefix should not be necessary?

Reply:
no problem, I was just following the convention that was there in the existing error handling code